### ISSUE TITLE
start TB with BT is sometimes starting multiple instances of TB's

### ISSUE THREAD

**Issue by JackDinn** [gyunaev/birdtray#24]
mainly when iv just rebooted my OS.
Using "start thunderbird when birdtray starts".

Im getting multiple instances of thunderbird started.

```
❰greg❙~/git/birdtray/src(git✱≠master)❱✔≻ ./birdtray 
Window ID found: 0
Starting Thunderbird as '/usr/bin/thunderbird'
Unread counter for /home/greg/.thunderbird/tzvg3gbn.default/Mail/smart mailboxes/Inbox.msf: 0
Window ID found: 0
Window ID found: 0
Window ID found: 0
Window ID found: 0
Window ID found: 0
Window ID found: 0
Starting Thunderbird as '/usr/bin/thunderbird'
Window ID found: 0
Window ID found: 0
[calBackendLoader] Using Thunderbird's builtin libical backend
Window ID found: 0
Window ID found: 0
Window ID found: 0
Starting Thunderbird as '/usr/bin/thunderbird'
Window ID found: 0
Window ID found: 0
Window ID found: 0
Window ID found: 0
Window ID found: 0
Window ID found: 0
Starting Thunderbird as '/usr/bin/thunderbird'
Window ID found: 0
Window ID found: 0
Window ID found: 0
Window ID found: 0
Window ID found: 0
Window ID found: 0
Starting Thunderbird as '/usr/bin/thunderbird'
Window ID found: 0
Window ID found: 0
Window ID found: 0
JavaScript error: chrome://calendar/content/calendar-views.xml, line 256: TypeError: this.rangeStartDate is null
Window ID found: 0
Window ID found: 0
Starting Thunderbird as '/usr/bin/thunderbird'
JavaScript error: chrome://global/content/bindings/tree.xml, line 1285: TypeError: this.parentNode is null


```

**Comment by gyunaev** [comment]
It doesn't start within 5 seconds? You can increase it to 10:

mThunderbirdRestartTime = QDateTime::currentDateTimeUtc().addSecs( 5 );

replace 5 by 10.

**Comment by gyunaev** [comment]
Because if you already have Thunderbird running before you started Birdtray, and you start another one, the one Birdtray starts exists immediately. I do not know then whether it just finished by itself (and I should not restart it), or the user closed it (and I should restart it).

**Comment by JackDinn** [comment]
Fair enough, iv adjusted the timer anyhow although i have not rebooted my system since then to test it loading as a startup. Should be ok though.

**Comment by gyunaev** [comment]
Happened to me as well :( need to rethink the start approach.

**Comment by lyndhurst** [comment]
I just want to comment because I think BirdTray is an essential tool, and it is doing a perfect job so far, except from this multiple instances management.

I personally do not have a problem anymore with boot time, it is not an absolute necessity for me, so I can disable it, and saving the kde session keeps BirdTray in the system tray anyway.

Running it on kubuntu 18.04.1, there is a few other sacrifices I have to make in order to use BirdTray. I built it from source, and did not install it through the linuxuprising ppa if that makes any difference.

I like to have tb pinned on my dock (latte), and a keyboard shortcut to quickly open it. 

If I launch tb while BirdTray is in the system tray, a second tb instance is launched, and (with my settings: all checked in the 'hiding' tab), BirdTray kills the newly launched instance, shows the red cross status for a sec, and launches a new tb instance again.

At that point, clicking on BirdTray has no effect anymore, and I have two thunderbird instances running in the background. All I can do, is quit, and, launch BirdTray again.

Maybe BirdTray should check if it is already running when launched, and if yes, show the tb hidden window instead of starting... That would not solve the dock icon situation though, this issue might need a plasmoid widget, or the if...then scenario described by the OP might be enough.

To summarize, if I use BirdTray, the only way to access tb without something crashing is double clicking the tray icon which is not always the most convenient to me.

Thank you very much for this great addition to thunderbird anyway.

**Comment by Kr1ss-XD** [comment]
FWIW, I notice the same behavior @lyndhurst describes as well, running current birdtray version from GitHub on Arch + Cinnamon.

As for the main issue (multiple instances opened at startup), I increased the `QDateTime::currentDateTimeUtc().addSecs()` argument to 10, and even tried 20 and 30 seconds, but sadly it didn't help.

Anyways, thx a lot for this great piece of software. Hope u'll get this fixed so it become even better. If there's need for some debugging or testing, I'd be more than happy to assist.

**Comment by lyndhurst** [comment]
Thanks for the fix.

I guess it applies to the multiple instances issue affecting mainly boot time. 
I just tried to build the latest version, I am not sure if the fix was included yet.

If yes, I can still see the same behavior when lauching TB while BT is already running along with TB.
No additional instance of TB is created, but, the tray icon becomes irresponsive to double clicks.

Let me know if I should open an issue with better description, I will be glad to help debug this if I can.
